### PR TITLE
Refactor agent plugin to support JDBC

### PR DESCRIPTION
**Pull request by jiangML**
For #22409

Changes proposed in this pull request:
  - Refactor agent plugin to support JDBC
---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: agent/api/src/main/java/org/apache/shardingsphere/agent/spi/boot/PluginBootService.java [hunk 1]
```diff
@@ -27,8 +27,9 @@ public interface PluginBootService extends AgentTypedSPI, AutoCloseable {
     
     /**
      * Start plugin boot service.
-     *
-     * @param pluginConfig plugin configuration
+     * 
+     * @param pluginConfig pluginConfig plugin configuration
```
Comment (RaigorJiang) [inline agent/api/src/main/java/org/apache/shardingsphere/agent/spi/boot/PluginBootService.java:31]: Duplicate pluginConfig

File: agent/core/src/main/java/org/apache/shardingsphere/agent/core/plugin/PluginBootServiceManager.java [hunk 2]
```diff
@@ -38,19 +38,20 @@ public final class PluginBootServiceManager {
     
     /**
      * Start all services.
-     *
+     *
```
Comment (RaigorJiang) [inline agent/core/src/main/java/org/apache/shardingsphere/agent/core/plugin/PluginBootServiceManager.java:41]: unnecessary space

File: agent/plugins/metrics/prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/collector/BuildInfoCollector.java [hunk 3]
```diff
@@ -32,23 +33,28 @@
...
  */
 @Slf4j
+@AllArgsConstructor
 public final class BuildInfoCollector extends Collector {
     
     private static final String PROXY_BOOTSTRAP_CLASS = "org.apache.shardingsphere.proxy.Bootstrap";
     
     private static final PrometheusWrapperFactory FACTORY = new PrometheusWrapperFactory();
     
+    private final boolean isRunWithProxy;
```
Comment (RaigorJiang) [inline agent/plugins/metrics/prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/collector/BuildInfoCollector.java:43]: What is the different between `isEnhancedForProxy`  and `isRunWithProxy`

File: agent/plugins/metrics/prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/collector/BuildInfoCollector.java [hunk 4]
```diff
@@ -32,23 +33,28 @@
...
         addMetric(artifactInfo.get(), getClass().getPackage());
-        try {
-            addMetric(artifactInfo.get(), Class.forName(PROXY_BOOTSTRAP_CLASS).getPackage());
-        } catch (final ClassNotFoundException ignored) {
-            log.warn("No proxy class find");
+        if (isRunWithProxy) {
+            try {
+                addMetric(artifactInfo.get(), Class.forName(PROXY_BOOTSTRAP_CLASS).getPackage());
+            } catch (final ClassNotFoundException ignored) {
+                log.warn("No proxy class find");
```
Comment (RaigorJiang) [inline agent/plugins/metrics/prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/collector/BuildInfoCollector.java:56]: `find` shoule be `found`.

File: agent/plugins/metrics/prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/collector/BuildInfoCollector.java (hunk 3 above)
Comment (jiangML) [reply agent/plugins/metrics/prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/collector/BuildInfoCollector.java:43]: I mean the same thing, let me adjust it.

File: agent/plugins/metrics/prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/collector/BuildInfoCollector.java (hunk 4 above)
Comment (jiangML) [reply agent/plugins/metrics/prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/collector/BuildInfoCollector.java:56]: ok

File: agent/api/src/main/java/org/apache/shardingsphere/agent/spi/boot/PluginBootService.java (hunk 1 above)
Comment (jiangML) [reply agent/api/src/main/java/org/apache/shardingsphere/agent/spi/boot/PluginBootService.java]: It did repeat that I modified it.

File: agent/core/src/main/java/org/apache/shardingsphere/agent/core/plugin/PluginBootServiceManager.java (hunk 2 above)
Comment (jiangML) [reply agent/core/src/main/java/org/apache/shardingsphere/agent/core/plugin/PluginBootServiceManager.java]: ok thanks, i will remove

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**